src/plugins/workspace/server/saved_objects/integration_tests/workspace_id_consumer_wrapper.test.ts

```diff
@@ -312,5 +318,128 @@ describe('workspace_id_consumer integration test', () => {
       expect(importWithWorkspacesResult.body.success).toEqual(true);
       expect(findResult.body.saved_objects[0].workspaces).toEqual([createdFooWorkspace.id]);
     });
+
+    it('get', async () => {
+      await clearFooAndBar();
+      const createResultFoo = await osdTestServer.request
+        .post(root, `/w/${createdFooWorkspace.id}/api/saved_objects/_bulk_create`)
+        .send([
+          {
+            ...dashboard,
+            id: 'foo',
+          },
+        ])
+        .expect(200);
+
+      const createResultBar = await osdTestServer.request
+        .post(root, `/w/${createdBarWorkspace.id}/api/saved_objects/_bulk_create`)
+        .send([
+          {
+            ...dashboard,
+            id: 'bar',
+          },
+        ])
+        .expect(200);
+
+      const getResultWithRequestWorkspace = await osdTestServer.request
+        .get(root, `/w/${createdFooWorkspace.id}/api/saved_objects/${dashboard.type}/foo`)
+        .expect(200);
+      expect(getResultWithRequestWorkspace.body.id).toEqual('foo');
+      expect(getResultWithRequestWorkspace.body.workspaces).toEqual([createdFooWorkspace.id]);
+
+      const getResultWithoutRequestWorkspace = await osdTestServer.request
+        .get(root, `/api/saved_objects/${dashboard.type}/bar`)
+        .expect(200);
+      expect(getResultWithoutRequestWorkspace.body.id).toEqual('bar');
+
+      await osdTestServer.request
```

Comment:
Can we add one more case that `get global objects within a workspace` won't throw any error? It is to ensure we won't introduce regression for data sources and config.

Reply:
Sure, I will add more cases